### PR TITLE
[Bug 2397]: added auth token in headers for download file in nyaymitra user

### PR DESCRIPTION
**Pull request by ashishbeehyv**
added the auth token in headers
https://github.com/pucardotorg/dristi/issues/2397

## Summary by CodeRabbit

- **New Features**
	- Introduced the `AuthenticatedLink` component for secure file downloads, ensuring user authentication via a token.
	- Updated the `DocViewerWrapper` to utilize the new `AuthenticatedLink` for handling download links.

- **Bug Fixes**
	- Improved error handling during file download requests to enhance user experience.

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/authenticatedLink.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import axios from "axios";
+
+const AuthenticatedLink = ({ t, uri, displayFilename = false }) => {
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/authenticatedLink.js:4]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/authenticatedLink.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const handleClick = (e) => {
+    e.preventDefault();
+
+    const authToken = localStorage.getItem("token");
+    axios
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/authenticatedLink.js:9]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/authenticatedLink.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      .then((response) => {
+        if (response.status === 200) {
+          const blob = new Blob([response.data], { type: "application/octet-stream" });
+          const blobUrl = URL.createObjectURL(blob);
+
+          window.open(blobUrl, "_blank");
+        } else {
+          console.error("Failed to fetch the PDF:", response.statusText);
+        }
+      })
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/authenticatedLink.js:25]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/authenticatedLink.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      .catch((error) => {
+        console.error("Error during the API request:", error);
+      });
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/authenticatedLink.js:28]: not needed

**Review by vaibhavct — APPROVED** [review]

**Pull request merged**